Added ORIGINAL CONTENT skeleton toys that chatter when you play with them

coiax opened the pull request:
I've done a ton of work today. A skele-ton.

:cl: coiax
rscadd: Toy fans, rejoyce! The famous skeleton brothers from the beloved game Belowyarn now have toys that you can win from the arcade machines!
/ :cl:

coiax committed:
Step one, aquire voices for skeletons

coiax committed:
Moved sounds to their own folder

coiax committed:
Added samples of up to ten phomemes long for papyrus

coiax committed:
Added samples of up to ten sans phomemes

coiax committed:
Add skeleton talking

Skeletons now produce audible sound effects representing their distinctive
speech.

coiax committed:
Skeletons now respect punctuation in their chatter

Except for semi-colons, due to HTML escapes of the ' character.

Cheridan commented:
understale is for losers

MrPerson commented:
4 years from now, will people still recognize this as a reference? Does it sound good to people who don't get the reference?

coiax commented:
I don't think it sounds awful. And people will probably recognise it as a reference, the sounds are pretty distinctive.

Cheridan commented:
Sorry but this is in violation of our reference policy.

coiax commented:
@RemieRichards DEFEND MY HONOUR

coiax commented:
> reference policy

duncathan commented on code/modules/mob/living/carbon/human/species_types.dm:
```diff
@@ -419,16 +419,116 @@ var/regex/lizard_hiSS = new("S+", "g")
+		if (word in punctuation)
+			letter_count += word
+			continue
+		letter_count += length(words[i])
```
replace everything below that first comment until here with this
```
var/global/regex/R
if(!R)
	R = regex("(\[\\l\\d]*)(\[^\\l\\d\\s])?", "g") //this regex does that for us!
var/list/L = list()
while(R.next <= length(message))
	R.Find(message)
	if(R.group[1])
		L += length(R.group[1])
	if(R.group[2])
		L += R.group[2]
```

PKPenguin321 commented:
this made me depressed
there was probably a lot of work put into this only for it to get shut down immediately
rip :disappointed:

coiax commented:
> this made me depressed
> there was probably a lot of work put into this only for it to get shut down immediately
> rip :disappointed:

Certainly sums up my feelings on the matter.

Cheridan commented:
Seeing as how this is just building onto an existing lame reference I guess I'll let it slide

Bawhoppen commented:
:-1: does not fit in the game AND is a reference to an even more shit game

coiax committed:
Lowered volume of skeleton phomemes

coiax committed:
Use Dunc's regex for parsing messages

CosmicScientist commented:
@PKPenguin321 could you put together the sounds on puush for the lazy people (me) to hear?

I'm automatically against this due to cringing at the sans and papyrus font reference maymay because I am anti-fun.

coiax commented:
@CosmicScientist added a link to the demo video in first post

xxalpha commented:
This sounds awful. The memememe was fine but this is just awful.

TheWiznard commented:
if you can't add them as actual voices add them as arcade prizes

GunHog commented:
+1 to the arcade prize idea. It is easier to deal with it, and I can bash someone's face for using it to pester people.

CosmicScientist commented:
I am anti-fun and therefore my opinion is obvious.

coiax commented:
Huh, hmm. Well, making some toy skeletons would be fun, and I'd be able to keep all the work I made.

Saegrimr commented:
Reference or not, its terrible sounding and jarring compared to everything else in the game.

At least the memememe speak was softer.

Shadowlight213 commented:
![](https://assets-cdn.github.com/images/icons/emoji/unicode/1f44e.png)
It sounds terrible

coiax committed:
Added sans and papyrus toy sprites

Spriting is hard. But I think I managed to crush the images down to fit
in our 32x32 box.

coiax committed:
Generic talking action figures

Talking action figures like the owl figure and the griffin figure are
now generic, and have a pool of messages to pick from. This allows us to
make our TOTALLY ORIGINAL skeleton toys.

coiax committed:
Added the toys to the prize list

No point making these if you can't get them.

coiax committed:
Moved skeleton vocalising code into toys

coiax committed:
Merge branch 'master' into voiced-skeletons

coiax committed:
Fixes runtimes and bugs with skeleton talk

Remaking the regex each time makes sure it gives output everytime,
and we've also put a check that the item in the list is actually
a number before calling max() that was causing a runtime previously.

RemieRichards commented on code/modules/mob/living/carbon/human/species_types.dm:
```diff
@@ -413,11 +413,10 @@
-	myspan = list(pick(SPAN_SANS,SPAN_PAPYRUS)) //pick a span and stick with it for the round
-
+	var/span = pick(SPAN_SANS,SPAN_PAPYRUS) //pick a span and stick with it for the round
+	myspan = list(span)
```
Why? My original form is better.

coiax committed:
Restored original form of skeleton span picking

Remie's was better.

coiax committed:
More messages for skeleton toys

I think that's probably enough to satisfy me.

coiax commented:
Well, I'm gonna call this as finished as it's gonna be.

RemieRichards commented on code/game/objects/items/toys.dm:
```diff
@@ -681,43 +682,129 @@
+
+	spawn(0)
+		// for each word of length N
+		for(var/i = 1, i <=letter_count.len, i++)
```
```for(var/i in 1 to letter_count.len)```

coiax committed:
Changed for loop to new style

ChangelingRain commented:
That's the disposal bin. Feel free to visit it any time.
